### PR TITLE
[irods/irods#5250] cmake: set project name and version number in project call (4-2-stable)

### PR DESCRIPTION
**Pull request by SwooshyCueb**
in service of irods/irods#5250

I couldn't get this repository to build in my docker container, but it got far enough along in the build process that I feel confident that I haven't broken anything

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -30,15 +31,15 @@ set(CMAKE_C_COMPILER ${IRODS_EXTERNALS_FULLPATH_CLANG}/bin/clang)
 set(CMAKE_CXX_COMPILER ${IRODS_EXTERNALS_FULLPATH_CLANG}/bin/clang++)
 set(CMAKE_EXE_LINKER_FLAGS_INIT "${CMAKE_EXE_LINKER_FLAGS} -stdlib=libc++")
 
-project(irods-s3 C CXX)
+project(irods-s3
```
Comment (trel) [inline CMakeLists.txt:34]: hmmm, perhaps `irods_resource_plugin-s3` ?

**Comment by JustinKyleJames** [comment]
> in service of irods/irods_5250
> 
> I couldn't get this repository to build in my docker container, but it got far enough along in the build process that I feel confident that I haven't broken anything

Without seeing the error, I would guess that the package irods-externals-libs34e684077-0 is not installed.  ?

**Comment by SwooshyCueb** [comment]
> Without seeing the error, I would guess that the package irods-externals-libs34e684077-0 is not installed. ?

I can't remember what the problem was with stable, but for master, clang complained about a bunch of undefined references to S3 stuff. I regenerated my docker container yesterday morning, so if new packages have been added recently, I should have them.

**Comment by JustinKyleJames** [comment]
I was able to build the plugin off of this pull request.

**Comment by trel** [comment]
excellent.  merging.

**Pull request merged**